ggircs-app/app/pages/xml-diff.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { Component } from "react";
+import { graphql } from "react-relay";
+import { Row, Col } from "react-bootstrap";
+import Button from "@button-inc/bcgov-theme/Button";
+import { xmlDiffQueryResponse } from "xmlDiffQuery.graphql";
+import { PageComponentProps } from "next-env";
+import DefaultLayout from "components/Layout/DefaultLayout";
+import { USER_GROUP } from "data/group-constants";
+import ReportSelector from "components/XmlDiff/ReportSelector";
+import DiffDetails from "components/XmlDiff/DiffDetails";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faExchangeAlt } from "@fortawesome/free-solid-svg-icons";
+import format from "xml-formatter";
+import RenderDiff from "components/XmlDiff/RenderDiff";
+// Use dynamic import for BootstrapSwitchButton (workaround for SSR bug where window is not defined)
+import dynamic from "next/dynamic";
+const BootstrapSwitchButton = dynamic(
+  () => import("bootstrap-switch-button-react"),
+  {
+    loading: () => <p>loading</p>,
+    ssr: false, // This line is important. It's what prevents server-side render
+  }
+);
+
+const ALLOWED_GROUPS = [...USER_GROUP];
+
+interface Props extends PageComponentProps {
+  query: xmlDiffQueryResponse["query"];
+}
+interface State {
+  leftSideId: Number;
+  leftSideReport: any;
+  rightSideId: Number;
+  rightSideReport: any;
+  isReversed: Boolean;
+  isCollapsed: Boolean;
+}
+export default class Index extends Component<Props, State> {
+  static allowedGroups = ALLOWED_GROUPS;
+  static isAccessProtected = true;
+  static query = graphql`
+    query xmlDiffQuery {
+      query {
+        session {
+          ...DefaultLayout_session
+        }
+        allReports {
+          edges {
+            node {
+              swrsReportId
+              latestSwrsReport {
+                submissionDate
+                ecccXmlFileByEcccXmlFileId {
+                  xmlFileName
+                  xmlFile
+                  ecccZipFileByZipFileId {
+                    zipFileName
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  `;
+
+  state = {
+    leftSideId: null,
+    leftSideReport: null,
+    rightSideId: null,
+    rightSideReport: null,
+    isReversed: false,
+    isCollapsed: false,
+  };
```

Comment:
do we want to move those to the query string?

Reply:
do we? The leftsideReport / rightSideReport variables are complex objects which contain the full report xml

Reply:
the ids and boolean could be moved though, especially the ids, as it would enable sharing the URL, or refreshing the page, and still end up with the same view

Reply:
variables to router work in issue #331